scipy/optimize/optimize.py

```diff
@@ -1465,7 +1465,8 @@ def _minimize_newtoncg(fun, x0, args=(), jac=None, hess=None, hessp=None,
             A = fhess(*(xk,) + args)
             hcalls = hcalls + 1
 
-        while numpy.add.reduce(numpy.abs(ri)) > termcond:
+        k2 = 0
+        while (numpy.add.reduce(numpy.abs(ri)) > termcond) and (k2 < maxiter):
```

Comment:
If it converges in order-N iterations *under reasonably weak conditions*, then I'd say all we need to do is to pick a multiple of N. Like, 10N or 50N or maybe keep it

```
    if maxiter is None:
        maxiter = len(x0)*200
```
?